apiserver/common/relationunitswatcher.go

```diff
@@ -9,28 +9,20 @@ import (
 	"github.com/juju/worker/v4/catacomb"
 
 	"github.com/juju/juju/core/watcher"
+	"github.com/juju/juju/domain/relation"
 	"github.com/juju/juju/rpc/params"
-	"github.com/juju/juju/state"
 )
 
-// RelationUnitsWatcher represents a state.RelationUnitsWatcher at the
+// RelationUnitsWatcher represents a relation.RelationUnitsWatcher at the
 // apiserver level (different type for changes).
 type RelationUnitsWatcher interface {
 	watcher.Watcher[params.RelationUnitsChange]
-
-	// Stop is needed to implement facade.Resource.
-	// Deprecated: use Kill and Wait instead.
-	Stop() error
-
-	// Err implements watcher.Errer.
-	// Deprecated: use Kill and Wait instead.
-	Err() error
 }
 
-// RelationUnitsWatcherFromState wraps a state-level
+// RelationUnitsWatcherFromDomain wraps a state-level
 // RelationUnitsWatcher in an equivalent apiserver-level one, taking
 // responsibility for the source watcher's lifetime.
```

Comment:
Might also be good to make this comment more specific. From what I understand all this wrapper does is change the emitted watcher even into a `params` struct. I may have missed something though.

Reply:
Technically it's a new watcher with different structs as the comment says.